code/game/gamemodes/objective.dm

```diff
@@ -421,6 +421,55 @@ GLOBAL_LIST(admin_objective_list) //Prefilled admin assignable objective list
 			return ..()
 	return SSshuttle.emergency.is_hijacked() || ..()
 
+/datum/objective/gimmick
+	name = "gimmick"
+	martyr_compatible = TRUE
+
+/datum/objective/gimmick/update_explanation_text()
+	var/selected_department = pick(list( //Select a department for department-based objectives
+		DEPT_SCIENCE,
+		DEPT_ENGINEERING,
+		DEPT_SECURITY,
+		DEPT_MEDICAL,
+		DEPT_SERVICE,
+		DEPT_SUPPLY,
+		DEPT_COMMAND
+	))
+	var/list/gimmick_list = list( //This list is ONLY for objectives that depend on a random selection
+		"Get everyone to hate [selected_department], through misdirection, slander, and sabotage.",
+		"Incite a strike in [selected_department].",
+		"Impede the work of those at [selected_department] while keeping those working in it alive.",
+		"Keep power & cameras cut in [selected_department].",
+		"Disguise yourself as a member of [selected_department] and sabotage them from within.",
+		"Vandalize [selected_department] as much as possible.",
+		"Destroy [selected_department] utterly and entirely. Deaths within the department are of no concern, but limit the destruction to [selected_department].", //Teeeeechnically, there is a 1/50 chance of getting any specific gimmick objective, making this one rarer than hijack
+		"Steal equipment critical for [selected_department] to function.",
+		"Attempt to kidnap a member of [selected_department] to take with you alive on an escape pod or a shuttle, for questioning at Syndicate HQ.",
+		"Hack all doors in [selected_department] to be shocked, bolted, or inaccessible by the AI"
+	)
+	if(target && target.current) //Only add these objectives to the pool if there are targets on the station
+		gimmick_list.Add(
+			"Construct an elaborate trap for [target.name].",
+			"Ensure [target.name] is demoted from their position.",
+			"Get [target.name] arrested for a crime they haven't committed.",
+			"Top-secret Syndicate intelligence suggests that [target.name] is responsible for scratching the paint on one of our CEOs' private cruisers. Make their life as miserable as possible without killing them.",
+			"Kidnap [target.name] and hold them for ransom!",
+			"Replace [target.name] completely, and try as hard as possible not to blow your cover.",
+			"Pin the blame on [target.name] for any and all crimes you commit, and try to convince security and the crew that it was them all along.",
+			"Attempt to kidnap [target.name] to take with you alive on an escape pod or shuttle, for questioning at Syndicate HQ.",
+			"Frame [target.name] for murdering you.",
+			"[target.name] knows corporate secrets. Interrogate them. Use force if they pretend to not know."
+		)
```

Comment:
Could be nicee to instead have it read gimmick files and replace some text such as %TARGET with target.name, so then all of these can be stored in the text file to with:
%TARGET knows corporate secrets...
Would make the code a bit cleaner.

Reply:
fixed, moved departmental and target specific objectives into their own text files and added a thing for it to select from them